File: src/nl/rubensten/texifyidea/structure/filter/DummyTreeElement.java
package nl.rubensten.texifyidea.structure.filter;

import com.intellij.ide.util.treeView.smartTree.TreeElement;
import com.intellij.navigation.ItemPresentation;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * @author Ruben Schellekens
 */
public enum DummyTreeElement implements TreeElement {

    INSTANCE;

    @NotNull
    @Override
    public ItemPresentation getPresentation() {
        return DummyPresentation.INSTANCE;
    }

    @NotNull
    @Override
    public TreeElement[] getChildren() {
        return TreeElement.EMPTY_ARRAY;
    }

    /**
     * @author Ruben Schellekens
     */
    private static class DummyPresentation implements ItemPresentation {

        private static final ItemPresentation INSTANCE = new DummyPresentation();

        @Nullable
        @Override
        public String getPresentableText() {
            return null;
        }

        @Nullable
        @Override
        public String getLocationString() {
            return null;
        }

        @Nullable
        @Override
        public Icon getIcon(boolean b) {
            return null;
        }
    }
}

File: src/nl/rubensten/texifyidea/structure/filter/DummyTreeElement.kt
package nl.rubensten.texifyidea.structure.filter

import com.intellij.ide.util.treeView.smartTree.TreeElement
import com.intellij.navigation.ItemPresentation
import javax.swing.Icon

/**
 * @author Ruben Schellekens
 */
enum class DummyTreeElement : TreeElement {

    INSTANCE;

    override fun getPresentation() = DummyPresentation

    override fun getChildren(): Array<out TreeElement> = TreeElement.EMPTY_ARRAY

    /**
     * @author Ruben Schellekens
     */
    object DummyPresentation : ItemPresentation {

        override fun getPresentableText(): String? = null

        override fun getLocationString(): String? = null

        override fun getIcon(b: Boolean): Icon? = null
    }
}
